ckanext/ytp/request/logic/action/update.py

```diff
@@ -49,7 +50,10 @@ def _process(context, action, data_dict):
     role = data_dict.get("role", None)
     if not mrequest_id:
         raise logic.NotFound
-    if role is not None and role != 'admin' and role != 'editor':
+
+    valid_roles = t.get_action('member_roles_list')(context, {})
+    valid_roles = [r['value'] for r in valid_roles] + [None]
```

Comment:
Why do we need `None` in this list?

Reply:
I believe that it should be possible to approve a request without specifying a role, i.e. accept which ever default role is already in the request.